content/blog/2016/2016-04-07-pipeline-for-runs-on-hardware.adoc

```diff
@@ -0,0 +1,258 @@
+There are several features in Pipeline, which simplify the development, but we still need a full-featured IDE.
+
+## Conclusions
+
+Jenkins is a powerful *automation framework*, which can be used in many areas.
+Even though Jenkins has no dedicated plugins for test runs on hardware, it provides many general-purpose "building blocks", which allow implementing almost any flow.
+That's why Jenkins is so popular in the hardware and embedded areas.
+
+Pipeline-as-code is an emerging technology, which should greatly simplify the implementation of complex flows.
+Currently it lacks integrations with particular Jenkins features, but hopefully this issue will be solved soon.
+
+If you develop new automation flows, consider Pipeline as one of possible approaches.
+
+## What's next?
+
+Jenkins automation server dominates in the HW/embedded area, but unfortunately there is not so much experience sharing for these use-cases. 
+So Jenkins community encourages everybody to share the experience in this area by writing docs and articles for Jenkins website and other resources.
+
+This is just a a first blog post on this topic.
+I am planning to provide more examples of Pipeline-as-code usage for Embedded and Hardware tests in the future posts.
+The next post will be about concurrency and shared resource management in Pipelines.
+
+I am also going to talk about running tests on hardware at the 
+link:https://www.eventbrite.com/e/accelerating-automotive-innovation-with-continuous-integration-delivery-tickets-20809772590[incoming Automotive event] in Stuttgart on April 26th.
+This event is being held by 
+link:https://www.cloudbees.com/[CloudBees], but there will be several talks addressing Jenkins open-source as well.
+
+If you want to share your experience about Jenkins usage in Hardware/Embedded areas, consider submitting a talk for the 
+link:https://jenkins-cfp.herokuapp.com/events/jenkins-world-2016[Jenkins World conference] or join/organize a 
+link:https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+Area+Meetup[Jenkins Area Meetup] in your city. 
+There is also a 
+link:http://www.meetup.com/Jenkins-online-meetup/[Jenkins Online Meetup].
+
+## Links
+
+Related articles and events:
+
+* link:/solutions/embedded/[HW/Embedded Solution page]
+* link:https://www.cloudbees.com/jenkins/juc-2015/presentations/JUC-2015-Europe-Jenkins-Based-CI-for-Nenashev.pdf[Jenkins-Based CI for Heterogeneous Hardware/Software Projects]
+* link:https://www.eventbrite.com/e/accelerating-automotive-innovation-with-continuous-integration-delivery-tickets-20809772590[Accelerating Automotive Innovation with Continuous Integration & Delivery] - meetup in Stuttgart
+
+Pipeline:
+
+* link:/pipeline[Pipeline-as-Code Solutions page]
+* link:https://speakerdeck.com/onenashev/spb-jenkins-meetup-number-1-jenkins-2-dot-0-i-pipeline-as-code-eng[Jenkins 2.0 and Pipeline-as-code overview]
+* link:https://github.com/jenkinsci/workflow-plugin/blob/master/TUTORIAL.md[Pipeline Tutorial]
```

Comment:
Yes, yes it has. Click on that link and you'll note that it redirects to https://github.com/jenkinsci/pipeline-plugin/blob/master/TUTORIAL.md